### ISSUE TITLE
[RFC] Consider dropping localization (i18n / gettext / .po files)

### ISSUE THREAD

**Issue by pevik** [iputils/iputils#271]
Localization code is broken on some places in `ping`.
We set C locale (`setlocale(LC_ALL, "C");`) and then continue using `_()`:
https://github.com/iputils/iputils/blob/master/ping/ping_common.c#L202
https://github.com/iputils/iputils/blob/master/ping/ping_common.c#L889
(see discussion in https://github.com/iputils/iputils/pull/265#issuecomment-624510692).

Before fixing it, I wonder if we want to keep locales (`USE_GETTEXT`). I don't think that these low level tools should be localized. We only have 2 translations: Japanese and Ukrainian.

It's default on, but not all distros use it:

* Debian / Ubuntu
Don't see it from packaging files, but experimentally tested that it use packages
https://salsa.debian.org/debian/iputils/-/tree/master/debian

* Fedora
https://src.fedoraproject.org/rpms/iputils/blob/master/f/iputils.spec
use it (`BuildRequires: gettext`)

* Gentoo
https://gitweb.gentoo.org/repo/gentoo.git/tree/net-misc/iputils/iputils-20190709.ebuild
`-DUSE_GETTEXT="$(usex nls true false)"`
looks to me they allow user to choose

* openSUSE
https://build.opensuse.org/package/view_file/network:utilities/iputils/iputils.spec?expand=1
don't use it (missing gettext build dependency)

**Comment by pevik** [comment]
@jsynacek @nmeyerhans @vapier @pavlix @okias WDYT?

**Comment by vapier** [comment]
i'm not a fan of forcing English everywhere.  if anything, the sorry state of things makes it sound like some effort should be put in to integrate with something like [TP](https://translationproject.org/).  i know such projects aren't a silver bullet, but i really hate to see giving up on l10n entirely.  it's just a vicious cycle if every project keeps waiting on everyone else first :(.

Gentoo allows users to control l10n everywhere, not just iputils.  this is useful for e.g. embedded builds that would never having user-facing UI.

**Comment by pevik** [comment]
@tomop-tg @yurchor FYI

**Comment by pevik** [comment]
IMHO if we keep it, it should be fixed issues with `C` locale and `_()`. I'd prefer to use https://weblate.org/. We have also been contacted by folks from https://gitlocalize.com.

Plus it'd help if somebody start to care about localization and drive the effort. (We need to look on outstanding bugs and make release + I have some plans for testing, which is IMHO more important, thus have no energy to drive this.)

**Comment by yurchor** [comment]
> @tomop-tg @yurchor FYI

Whatever developers want. It always better to have excellent application than having a moderate quality application with a few translations.

Just wanted to point that Ubuntu translates iputils downstream via Launchpad:

https://translations.launchpad.net/ubuntu/focal/+source/iputils/+pots/iputils

**Comment by tomop-tg** [comment]
In Japan, some people have a negative reaction to English sentences. I hope this software works for those people as well.

**Comment by okias** [comment]
@tomop-tg question is, if there is someone from Japan, who is willing to contribute to iputils to make this solid.

As you wrote, there is most likely value for Japan users, but someone needs to care of it.

**Comment by vapier** [comment]
@okias, @pevik said there are Japanese translations available already

it's unreasonable to ask translators to maintain the underlying infrastructure in the code though (i.e. how we call l10n APIs including gettext)

**Comment by pevik** [comment]
OK, I see there is some requirement for translations.
I'll happily maintain translation code, but what would help me if somebody drive the translation effort. e.g. choose some webUI (weblate or any other which he likes) and take Ubuntu effort (the partial German translations).

**Comment by yurchor** [comment]
> Anybody volunteering to run upstream translates in some translate service? E.g. https://weblate.org or https://gitlocalize.com?

I can do this.

**Comment by pevik** [comment]
Thank you! Could you please also check Launchpad and take the output from there? (last time I saw German translation). We could create wiki page and/or doc in source code to notify the others).
BTW sometimes during summer holiday I might find a bit more time to look on the tickets and suggest what should be fixed before release. It'd be nice to have release during summer.

**Comment by yurchor** [comment]
> Thank you! Could you please also check Launchpad and take the output from there? (last time I saw German translation). We could create wiki page and/or doc in source code to notify the others).
> BTW sometimes during summer holiday I might find a bit more time to look on the tickets and suggest what should be fixed before release. It'd be nice to have release during summer.

E-m-m... What should be checked from LP? Ubuntu will translate your package independently anyway (it is enough to put a POT in the package to trigger this). It is traditionally up to the LP translation teams to backport translations upstream (if way want). See GNOME, GIMP, binutils, util-linux, coreutils, cryptsetup, etc. Ubuntu team will not do this.

**Comment by pevik** [comment]
I just hoped we could get that German translation. It's here: https://translations.launchpad.net/ubuntu/focal/+source/iputils/+pots/iputils/de/+translate (link from https://translations.launchpad.net/ubuntu/focal/+source/iputils/+pots/iputils). I mean don't wait for anybody to contribute it, just take what has been already translated. They haven't contributed the result up now, why would they bother now?
But, clearly I don't know how translating process works, sorry if I'm asking for something what is meaningless.

**Comment by pevik** [comment]
BTW I looked into sources and haven't found that German POT.

**Comment by yurchor** [comment]
And if I dare to ask what the option is preferable for iputils developers?

1. Hosted Weblate (I do not know if Michal still accepts free projects, there are too many of them these days).
2. GitLocalize (there are doubts that the project can survive in the long run).
3. Fedora's Weblate (the team is uncoordinated now). https://translate.fedoraproject.org/
4. Mageia's Transifex (non-FOSS, I'm a coordinator). https://www.transifex.com/MageiaLinux/mageia/dashboard/
5. LP (stagnating for a long time, but still works).
6. Some other options?

**Comment by yurchor** [comment]
> BTW I looked into sources and didn't find that German POT.

They patch the sources with their (LP) translations.

**Comment by yurchor** [comment]
Ping?

**Comment by pevik** [comment]
@yurchor I'm sorry for late reply. I'd prefer Weblate. But generally whatever works for you and produces translations :).

**Comment by yurchor** [comment]
> @yurchor I'm sorry for late reply. I'd prefer Weblate. But generally whatever works for you and produces translations :).

@pevik Ok. Thanks. Here is our new translation page on Fedora's Weblate:

https://translate.fedoraproject.org/projects/iputils/iputils/

If you do not want to manage translations by yourself, you can add me to the project and I will take care of translations. I promise that I will not touch any byte of code except for the translations.

**Comment by pevik** [comment]
> @pevik Ok. Thanks. Here is our new translation page on Fedora's Weblate:
> 
> https://translate.fedoraproject.org/projects/iputils/iputils

@yurchor great, thanks! We could update german translation :) (https://translate.fedoraproject.org/download/iputils/iputils/?format=zip).

> If you do not want to manage translations by yourself, you can add me to the project and I will take care of translations. I promise that I will not touch any byte of code except for the translations.

@okias, can you please add yurchor to the project?

**Comment by pevik** [comment]
@yurchor I'm working on new release. Should we add incomplete translations? French, Chinese, Portuguese? Or just fixes for Japanese?

**Comment by yurchor** [comment]
> @yurchor I'm working on new release. Should we add incomplete translations? French, Chinese, Portuguese? Or just fixes for Japanese?

@pevik Sure. Please add incomplete translations. I have asked Fedora teams to translate the package [1]. People want to see their translations. ;)

If you have some release date and a deadline I can ask for translations once more to have better coverage.

[1] https://lists.fedoraproject.org/archives/list/trans@lists.fedoraproject.org/thread/ULYD4WKBQWCYO2SNJ734VVU3IYRPCDT3/

**Comment by pevik** [comment]
To be honest, I'd like to release this week (I'll ping you in the pull request). But as there will be some bug fixes afterwards, I'll expect some bugfix release in few weeks.

The only way to update translations is to download the archive https://translate.fedoraproject.org/download/iputils/?format=zip ?
(There is no git repository for these translations I could just use to merge that translations?)

**Comment by yurchor** [comment]
> The only way to update translations is to download the archive https://translate.fedoraproject.org/download/iputils/?format=zip ?

Not the only one. But it seems we decided to do this way (see above).

> (There is no git repository for these translations I could just use to merge that translations?)

Yep. Greedy IBM/Red Hat does not want to buy another hard disk for their server and we have a resource shortage (I'm not joking). ;)

**Comment by pevik** [comment]
Thanks for info, I'll update that manually (unless @okias gives you permission to git repository so that you can manage it).

**Comment by okias** [comment]
@pevik for now I think enough people has push permission into repository, but in case someone need to regularly uodate translations and it would become burden to accept PR, then I'll give them access 😄
